src/Functions/FunctionsJSON.cpp

```diff
@@ -623,24 +623,32 @@ class JSONTypeImpl
     static bool insertResultToColumn(IColumn & dest, const Element & element, std::string_view)
     {
         UInt8 type;
-        if (element.isInt64())
-            type = 'i';
-        else if (element.isUInt64())
-            type = 'u';
-        else if (element.isDouble())
-            type = 'd';
-        else if (element.isBool())
-            type = 'b';
-        else if (element.isString())
-            type = '"';
-        else if (element.isArray())
-            type = '[';
-        else if (element.isObject())
-            type = '{';
-        else if (element.isNull())
-            type = 0;
-        else
-            return false;
+        switch (element.type())
+        {
+            case ElementType::INT64:
+                type = 'i';
+                break;
+            case ElementType::UINT64:
+                type = 'u';
+                break;
+            case ElementType::DOUBLE:
+                type = 'd';
+                break;
+            case ElementType::STRING:
+                type = '"';
+                break;
+            case ElementType::ARRAY:
+                type = '[';
+                break;
+            case ElementType::OBJECT:
+                type = '{';
+                break;
+            case ElementType::NULL_VALUE:
+                type = 0;
+                break;
```

Comment:
The return of `type()` is already mapped to the same values of the `simdjson` implementation, so I suspect the compiler will replace any switches on this with just the result of the type() call, if it was inlined, at least for the simdjson specialization, but I haven't verified it.
I would like to preserve that, in the general case. Here, most enum member values also match, except for null (`0` vs `n`), but I would still keep this as it, because this is the implementation of `JSONType`, but the call to `type()` happens in a lot of other places, in hotter code paths as well.